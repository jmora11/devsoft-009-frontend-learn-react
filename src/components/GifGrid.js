import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data: imagenes, loading } = useFetchGifs(category);

    return (
        <>
            <h1 className='animate__animated animate__bounce'>{category}</h1>

            { loading && <p>Loading...</p> }

            <div className='card-grid'>
                {
                    imagenes.map( img =>
                        <GifGridItem
                            key={ img.id }    
                            { ...img }
                        />  
                    )
                }
            </div>
        </>
    )
};
