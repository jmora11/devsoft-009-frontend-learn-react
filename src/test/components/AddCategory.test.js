import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from "enzyme";
import { AddCategory } from '../../components/AddCategory';

describe('Add Category', () => {

    const setCategories = jest.fn();

    let wrapper = shallow( <AddCategory setCategories={ setCategories } /> );

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories } /> );
    });

    test('debe mostrar correctamente el snapshot', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Demon Slayer';
        input.simulate('change', { target: { value } });
    });

    test('no debe postear la información onsubmit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect( setCategories ).not.toHaveBeenCalled();
    });

    test('debe postear la información onsubmit', () => {
        const value = 'Demon Slayer';
        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect( setCategories ).toHaveBeenCalledTimes(1);
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function));
        expect( wrapper.find('input').prop('value') ).toBe('');
    });

});