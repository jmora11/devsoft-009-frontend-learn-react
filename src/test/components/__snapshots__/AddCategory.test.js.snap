// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Category debe mostrar correctamente el snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="categori"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
